core_lib/src/interface/editor.cpp

```diff
@@ -914,17 +914,42 @@ void Editor::updateObject()
     emit updateLayerCount();
 }
 
-bool Editor::importBitmapImage(const QString& filePath, int space)
+Status Editor::importBitmapImage(const QString& filePath, int space)
 {
     QImageReader reader(filePath);
 
     Q_ASSERT(layers()->currentLayer()->type() == Layer::BITMAP);
     auto layer = static_cast<LayerBitmap*>(layers()->currentLayer());
 
+    Status status = Status::OK;
+    DebugDetails dd;
+    dd << QString("Raw file path: %1").arg(filePath);
+
     QImage img(reader.size(), QImage::Format_ARGB32_Premultiplied);
     if (img.isNull())
     {
-        return false;
+        QString format = reader.format();
+        if (!format.isEmpty())
+        {
+            dd << QString("QImageReader format: %1").arg(format);
+        }
+        dd << QString("QImageReader ImageReaderError type: %1").arg(reader.errorString());
+
+        QString errorDesc;
+        switch(reader.error())
+        {
+        case QImageReader::ImageReaderError::FileNotFoundError:
+            errorDesc = tr("File not found at path \"%1\". Please check if image is present and try again.").arg(filePath);
+            break;
+        case QImageReader::UnsupportedFormatError:
+            errorDesc = tr("This image format is not supported. Please try converting it to one of the following formats and try again:\n%1")
+                        .arg((QString)reader.supportedImageFormats().join(", "));
+            break;
+        default:
+            errorDesc = tr("We ran into an error while reading this image. Please check if it is valid and try again.");
+        }
```

Comment:
@MrStevns Thank you for your feedback! I've gone through and simplified/removed punctuations from the error titles and descriptions. Do you think I should revise these error descriptions to be a bit more direct?

Reply:
Hi. I think the descriptions could benefit from having the **error** messages with a bit more concise language, and the **action** the user is supposed to do could be a bit more direct and specific. 

From experience, even if it's contextually obvious the text is referring to the currently imported image, sometimes users can make incorrect choices by not having the description be direct. That is if they even read them instead of pressing thee [OK] button 😅 

That said,  for those who read, I would personally modify these in the following way:

```cpp
case QImageReader::ImageReaderError::FileNotFoundError:
            errorDesc = tr("File not found at \"%1\". Please check the image is present in the specified file location and try again.").arg(filePath);
            break;
        case QImageReader::UnsupportedFormatError:
            errorDesc = tr("Image format is not supported. Please convert the image file to one of the following formats and try again:\n%1")
                        .arg((QString)reader.supportedImageFormats().join(", "));
            break;
        default:
            errorDesc = tr("An error has occurred while reading the image. Please check the file has a valid image format and try again.");
        }
```

What do you guys think?

Reply:
I like Jose's suggested changes, particularly the more active voice in the second error message. I would make a couple slight modifications though:
- "Please check the image is present **at the specified location** and try again."
- "An error has occurred while reading the image. Please check **that** the file **is a valid image** and try again."

Reply:
Thank you both for the help! I've just updated the messages to reflect the suggested changes.